client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -4,25 +4,43 @@ import { get, includes } from 'lodash';
 import PropTypes from 'prop-types';
 import { Component } from 'react';
 import { connect } from 'react-redux';
+import { emailFormEventEmitter } from 'calypso/me/account/account-email-field';
 import {
 	verifyEmail,
 	resetVerifyEmailState,
 } from 'calypso/state/current-user/email-verification/actions';
 import { getCurrentUserEmail } from 'calypso/state/current-user/selectors';
-
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
 import './style.scss';
+import getUserSettings from 'calypso/state/selectors/get-user-settings';
+import isPendingEmailChange from 'calypso/state/selectors/is-pending-email-change';
+import { setUserSetting } from 'calypso/state/user-settings/actions';
+import { saveUnsavedUserSettings } from 'calypso/state/user-settings/thunks';
 
```

Comment:
this is the file for the dialog/modal. Where we have:

1. updated it to close on route change (reasons noted in description).
2. updated it to handle resending an email for a pending email change (as opposed to only verification).
3. updated it to handle the case of already being on /me/account where the previous anchor led.